### PR TITLE
Update GeographyPointValue equals method

### PR DESCRIPTION
**Pull request by wolffcm**
It wasn't smart about the poles or longitude [+-]180.

### CODE REVIEW

File: src/frontend/org/voltdb/types/GeographyPointValue.java [hunk 1]
```diff
@@ -144,12 +145,18 @@ public boolean equals(Object o) {
         }
 
         GeographyPointValue that = (GeographyPointValue)o;
+        if (this == that) {
+            return true;
+        }
+
+        GeographyPointValue normThis = normalizeLngLat(getLongitude(), getLatitude());
+        GeographyPointValue normThat = normalizeLngLat(that.getLongitude(), that.getLatitude());
```
Comment (BillWhite) [inline src/frontend/org/voltdb/types/GeographyPointValue.java]: Why use the normalized coordinates here?  Shouldn't the coordinates already be normalized?

Comment (wolffcm) [inline src/frontend/org/voltdb/types/GeographyPointValue.java]: Hmm, they will be normalized but not canonicalized.  I should have a function called canonicalizeLngLat that sets longitude to 0 at the poles, etc.  This need not be done in the normalize function, as it does in my commits.

I'll try again.

Comment (BillWhite) [inline src/frontend/org/voltdb/types/GeographyPointValue.java]: Normalizing converts (x, 90) to (0, 90) and (x, -90) to (0, -90), right?

Comment (BillWhite) [inline src/frontend/org/voltdb/types/GeographyPointValue.java]: I think what you have done is right.  The coordinates should be in range.  So all you will be doing with normalizeLngLat is fixing up lat=+/-90.  That seems right to me.

**Comment by BillWhite** [comment]
What did you do about the normalization tests?  Maybe we should just do away with them.  I'm not sure they are giving up much.

Other than that, it seems as if you have gotten everything right, and the tests seem complete.

**Comment by BillWhite** [comment]
The lng/lat normalization tests were auto-generated by the main in TestGeographyPointNormalization. It looks like you changed them by hand.  Perhaps the main routine there should do the right thing when Math.abs(lat) == 90.0.

File: src/frontend/org/voltdb/types/GeographyPointValue.java (hunk 1 above)
Comment (wolffcm) [inline src/frontend/org/voltdb/types/GeographyPointValue.java]: We also force a -180 longitude coordinate to be +180.

I was thinking that we should make a distinction between *normalizing*, that is, forcing coordinates to be in range with modulo-like arithmetic, and *canonicalizing*, forcing different representations of the same point to be the same.

Maybe that's not needed though.  If you're okay with this, I won't sweat it.

**Comment by wolffcm** [comment]
I didn't edit by hand, I re-auto-generated the TestGeographyPointNormalization tests, since they just use normalizeLngLat to generated expected values.

**Comment by wolffcm** [comment]
Merged externally.

**Pull request closed without merging**